### PR TITLE
issue 1: Application Controller + tests

### LINKED ISSUE #1: Deployment Environment
#### As a developer I want a continuous deployment environment setup for my new Rails application.

__Story__:

Hi! I’ve heard you want to build an image sharing website using
ruby-on-rails. That’s wonderful! You have quite the adventure ahead of you
while you build out this working site.

To help you get started, I have already created a source repository for
you. This repository, on GitHub, contains the source code, and the list of
issues, including this one, that you will work on as you build out the features
for your website.

While you build out features, you want to ensure you do not break
anything. Thus with each feature you write, you will also write the necessary
model, controller, and flow tests. While you should always run your tests
locally before committing code, we have configured CircleCI as our automated
continuous integration system; that is, CircleCI will run all the tests on a
branch whenever you push a modification to it on GitHub.

Finally, you have a complete deployment pipeline set up on Heroku. With this
pipeline, there are three steps used for testing your application. Review
applications are automatically created whenever you make a pull request. There
can be as many review applications as you have open pull requests. This step
enables you to perform quality assurance (QA) tests while you are developing a
feature. The second step, staging, is used to ensure no bugs are introduced as
the result of a feature branch merge into master. The staging server is as
close of a replica to your production server there is, so it provides a great
way to ensure that changes will not negatively impact your customers. The
staging server is automatically deployed whenever the master branch is updated
and the automated tests pass. The last step is production. This step contains
your production application, and the goal is for that server to have zero
unscheduled downtime – all of the other steps work toward that effort.

Please keep in mind that each feature delivers a modicum of value to either
your users, or to aid in your development of the website. You want to deliver
value to your users fast, so some features, like authentication, are saved
until your users really need them.

To begin with this story, please verify that things are set up for you by
checking off each item below. If anything does not appear to work, it’s up to
you to fix it and make it work.

I encourage you to read the discussion topics and keep them in mind when
working on the story. You should be able to provide answers to them when the
story is complete.

The references might provide insight into the discussion topics, help you
complete the story, or simply provide more information that you might be
interested in. It is not expected that you read through all the references --
you can if you want -- however, it might be a good idea to scan through them
before getting started with a story.

Happy coding!

__Acceptance criteria__:
- [x] [GitHub](https://github.com/AppFolioOnboarding) is used for source
  control.
- [x] The project's version of both Ruby and Rails are used both locally and on
  Heroku.
- [x] The landing page, `/`, provides some content.
- [x] Every available controller action has corresponding tests.
- [x] [CircleCI](https://circleci.com/) is used for Continuous Integration.
- [x] Heroku is used to deploy the application.
- [x] Pull requests are automatically deployed as review applications.
- [x] The staging application automatically deploys whenever the master branch
  is updated.
- [x] The staging application automatically deploys only after all tests pass
  on CircleCI.
- [x] The staging application deployment can be promoted to the production
  application deployment.
- [x] [Rubocop](https://github.com/bbatsov/rubocop) is used to help with code
  style consistency.

__Discussion Topics__:
- [x] Where can we discover how long it takes to return an HTTP response?
- [x] When should we commit our code?

__References__:
* https://rvm.io/rvm/install#try-out-your-new-rvm-installation
* http://guides.rubyonrails.org/getting_started.html
* http://guides.rubyonrails.org/testing.html#functional-tests-for-your-controllers
* https://devcenter.heroku.com/articles/getting-started-with-rails5
* https://devcenter.heroku.com/articles/pipelines
* https://github.com/bbatsov/ruby-style-guide

### PR DESCRIPTION
**Pull request by yasz24**
Closes #1

### CODE REVIEW

File: app/views/layouts/application.html.erb [hunk 1]
```diff
@@ -23,6 +23,6 @@
     <script src="https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0-beta.2/js/bootstrap.min.js" integrity="sha384-alpBpkh1PFOepccYVYDB4do5UnbKysX5WZXm3XxPqe5iKTfUKjNkCk9SaVuEZflJ" crossorigin="anonymous"></script>
 
     <%= javascript_include_tag 'application' %>
-    <%= javascript_pack_tag 'application' %>
+    <% #javascript_pack_tag 'application' %>
```
Comment (choruk) [inline app/views/layouts/application.html.erb]: Is this change intentional?

File: test/controllers/application_controller_test.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+require 'test_helper'
+
+class ApplicationControllerTest < ActionDispatch::IntegrationTest
+  test "application controller" do
+    get "/"
+    assert_response :success
+    assert_select "title", {:count=>0, :text=>"Hello World!"}
```
Comment (choruk) [inline test/controllers/application_controller_test.rb]: I'm not sure if this assertion is doing what you want. As written, it is asserting that the text `Hello World!` appears 0 times in the title of the page. I think what you want is to assert that the text `Hello World!` appears 1 time in the body of the page. The way `assert_select` works is that it accepts a css selector (so it can be the name of an HTML tag, a class name, an ID, or some combination of those). Your current view file is not rendering the text `Hello World!` in any html tags, so it will be difficult to assert on it (although ultimately all of your content will be inside of the `body` tag, so that is always an option). I'd recommend updating this assertion such that the test still passes if you pass `count: 1, text: 'Hello World!'` as the options. If you can't figure it out, let me know and we can pair on it.

File: test/controllers/application_controller_test.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+require 'test_helper'
+
+class ApplicationControllerTest < ActionDispatch::IntegrationTest
+  test "application controller" do
+    get "/"
+    assert_response :success
```
Comment (choruk) [inline test/controllers/application_controller_test.rb]: This assertion is actually verifying the HTTP status code that gets returned. By passing the symbol `:success`, you are asserting that the HTTP status code is any value from 200-299 ([more info on that here](https://api.rubyonrails.org/v5.2.3/classes/ActionDispatch/Assertions/ResponseAssertions.html)). In this case, we know the HTTP status code will be 200, so we can make this assertion a bit more strict. While you can always pass an integer value to `assert_response`, the mapping from symbol name to integer can be found in [the `Rack::Utils::HTTP_STATUS_CODES` constant](https://www.rubydoc.info/gems/rack/Rack/Utils) (so in the case of status code 200, the symbol is `:ok`).

File: app/views/layouts/application.html.erb (hunk 1 above)
Comment (yasz24) [reply app/views/layouts/application.html.erb]: yea, it was something that allen suggested

Comment (choruk) [reply app/views/layouts/application.html.erb]: Ok in general I would say we should just delete un-necessary code instead of commenting it out. Not a big deal right now, but keep that in mind going forward.
